File: enum/code.py
from enum import Enum

# Simple Enum
class OrderStatus(Enum):
  PLACED = "PLACED"
  CONFIRMED = "CONFIRMED"
  SHIPPED = "SHIPPED"
  DELIVERED = "DELIVERED"
  CANCELLED = "CANCELLED"

# Usage
status = OrderStatus.SHIPPED

if status == OrderStatus.SHIPPED:
    print("Your package is on the way!")

class Coin(Enum):
    PENNY = 1
    NICKEL = 5
    DIME = 10
    QUARTER = 25
    
    # constructor
    def __init__(self, value):
        self.coin_value = value
    
    def get_value(self):
        return self.coin_value
    
total = Coin.DIME.get_value() + Coin.QUARTER.get_value()  # 35
print(f"Total Coins: {total}")

File: enum/code.ts
{
  enum OrderStatus {
    Pending = "PENDING",
    Shipped = "SHIPPED",
    Delivered = "DELIVERED",
    Cancelled = "CANCELLED",
  }

  const status = OrderStatus.Shipped;
  console.log(`The order status is: ${status}`);

  class Coin {
    // Static readonly instances (enum-like behavior)
    static readonly Penny = new Coin(1);
    static readonly Nickel = new Coin(5);
    static readonly Dime = new Coin(10);
    static readonly Quarter = new Coin(25);

    private readonly _value: number;

    private constructor(value: number) {
      this._value = value;
    }

    getValue(): number {
      return this._value;
    }

    static getAllCoins(): Coin[] {
      return [Coin.Penny, Coin.Nickel, Coin.Dime, Coin.Quarter];
    }
  }

  const total: number = Coin.Dime.getValue() + Coin.Quarter.getValue();
  console.log(`Total value of Dime and Quarter: ${total} cents`);
  console.log("All Coins: ", Coin.getAllCoins());
}
